security/voters.rst

```diff
@@ -31,6 +31,22 @@ in the application, which can be: affirmative, consensus or unanimous.
 For more information take a look at
 :ref:`the section about access decision managers <components-security-access-decision-manager>`.
 
+.. tip::
+
+    The example used throughout this page features just two routes (``post_show`` and ``post_edit``).
+    However, the advantage of voters is that you can reuse them in *many* places and centralize
+    all permission logic. If you don't reuse permissions or the rules are basic, you instead
+    might want to do the check in the controller directly and throw an ``AccessDeniedException``
+    to create the correct response::
```

Comment:
I'd say we move this to the VERY top of this chapter - like the very first words. Then we can say something simpler:

```rst
If you need to deny access in a controller based on custom rules - perhaps only the "owner" of an object can access a page - you can always put that logic directly in your controller::

    ... the code

But if you need to repeat this logic in multiple places, a better option is to centralize that logic with a custom voter.
```

And then it would go into the current beginning "In order to use voters".

I think you're right to talk about this - so why not discuss the easy option (and when it might not work) immediately? :)

Reply:
I tried my best :-)